scripts/download-resources.mjs

```diff
@@ -101,54 +111,211 @@ export async function download(url, destPath, expectedSHA = '', overwrite = fals
+  if (onWindows) {
+    return await downloadZip(`${ kuberlrBaseURL }.zip`, finalKuberlrSHA, 'kuberlr', kuberlrPlatformDir);
+  }
 
+  return await downloadTarGZ(`${ kuberlrBaseURL }.tar.gz`, finalKuberlrSHA, 'kuberlr', kuberlrPlatformDir);
+}
+
+export default async function main() {
   fs.mkdirSync(binDir, { recursive: true });
 
-  // Download Kubectl
-  const kubeVersion = '1.20.7';
-  const kubectlURL = `https://dl.k8s.io/v${ kubeVersion }/bin/${ kubePlatform }/amd64/${ exeName('kubectl') }`;
+  const kuberlrVersion = '0.3.2';
+  const kuberlrBase = `https://github.com/flavio/kuberlr/releases/download/v${ kuberlrVersion }`;
+  const kuberlrBaseURL = `${ kuberlrBase }/kuberlr_${ kuberlrVersion }_${ kubePlatform }_amd64`;
+  const kuberlrPlatformDir = `kuberlr_${ kuberlrVersion }_${ kubePlatform }_amd64`;
+  const allKuberlrSHAs = await getResource(`${ kuberlrBase }/checksums.txt`);
+  const kuberlrSHA = allKuberlrSHAs.split(/\r?\n/).filter(line => line.includes(`kuberlr_${ kuberlrVersion }_${ kubePlatform }_amd64`));
+
+  switch (kuberlrSHA.length) {
+  case 0:
+    throw new Error(`Couldn't find a matching SHA for [kuberlr_${ kuberlrVersion }_${ kubePlatform }-amd64] in [${ allKuberlrSHAs }]`);
+  case 1:
+    break;
+  default:
+    throw new Error(`Matched ${ kuberlrSHA.length } hits, not exactly 1, for platform ${ kubePlatform } in [${ allKuberlrSHAs }]`);
+  }
+  const finalKuberlrSHA = kuberlrSHA[0].split(/\s+/, 1)[0];
+  const kuberlrPath = await downloadKuberlr(kuberlrBaseURL, finalKuberlrSHA, kuberlrPlatformDir, onWindows);
+
+  // Download Kubectl, either into the bin dir, or into kuberlr's directory of versioned kubectl's
+  const kubeVersion = (await getResource('https://dl.k8s.io/release/stable.txt')).trim();
+  const kubectlURL = `https://dl.k8s.io/${ kubeVersion }/bin/${ kubePlatform }/amd64/${ exeName('kubectl') }`;
   const kubectlSHA = await getResource(`${ kubectlURL }.sha256`);
-  const kubectlPath = path.join(binDir, exeName('kubectl'));
+  // if there's no kuberlr, use a non-symlinked kubectl
+  const binKubectlPath = path.join(binDir, exeName('kubectl'));
+  let needToRelink = true;
+  const kuberlrDir = path.join(await findHome(), '.kuberlr', `${ kubePlatform }-amd64`);
+  const pathParts = path.parse(binKubectlPath);
+  // let kuberlr manage different versions of kubectl
+  const managedKubectlPath = path.join(kuberlrDir, `${ pathParts.name }${ kubeVersion.replace(/^v/, '') }${ pathParts.ext }`);
 
-  await download(kubectlURL, kubectlPath, kubectlSHA);
+  // If kubectlPath is a symlink delete it before continuing
+  // There is no kuberlr, so install a real version of kubectl in .../bin
+  // If there's a symlink currently there we need to remove it or we'll overwrite kuberlr
+  try {
+    const binKubectlStat = await fs.promises.lstat(binKubectlPath);
+
+    if (binKubectlStat.isSymbolicLink()) {
+      if (kuberlrPath) {
```

Comment:
This is always true.

Reply:
Simplified code no longer has this code path.